core/src/main/java/tech/pegasys/web3signer/core/service/http/handlers/signing/eth2/Eth2SignForIdentifierHandler.java

```diff
@@ -213,7 +216,19 @@ private UInt64 toUInt64(final tech.pegasys.teku.infrastructure.unsigned.UInt64 u
   }
 
   private void respondWithSignature(final RoutingContext routingContext, final String signature) {
-    routingContext.response().putHeader(CONTENT_TYPE, TEXT_PLAIN_UTF_8).end(signature);
+    final String acceptableContentType = routingContext.getAcceptableContentType();
+    final String contentType;
+    final String body;
+
+    LOG.trace("Acceptable Content Type {}", acceptableContentType);
+    if (Objects.equal("application/json", acceptableContentType)) {
+      contentType = JSON_UTF_8;
+      body = new JsonObject().put("signature", signature).encode();
+    } else {
+      contentType = TEXT_PLAIN_UTF_8;
```

Comment:
Does this default to text/plain? If I send a request to Accept: */* will this return the plain text response?

Reply:
For `Accept` header with values `*/*` and `application/json` the `routingContext.getAcceptableContentType();`  returns `application/json`. If Accept header is any thing other than */*, application/json and text/<?>, the openapi router factory kicks in and returns a 403 I think (invalid media type).